x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/LabelFieldProducer.java

```diff
@@ -167,4 +169,47 @@ public void write(XContentBuilder builder) throws IOException {
             }
         }
     }
+
+    static final class FlattenedLastValueFieldProducer extends LabelLastValueFieldProducer {
+
+        FlattenedLastValueFieldProducer(String name) {
+            super(name);
+        }
+
+        @Override
+        public void collect(FormattedDocValues docValues, int docId) throws IOException {
+            if (isEmpty() == false) {
+                return;
+            }
+            if (docValues.advanceExact(docId) == false) {
+                return;
+            }
+
+            int docValuesCount = docValues.docValueCount();
+            assert docValuesCount > 0;
+            isEmpty = false;
+            List<BytesRef> values = new ArrayList<>(docValuesCount);
+            for (int i = 0; i < docValuesCount; i++) {
+                values.add(new BytesRef(docValues.nextValue().toString()));
```

Comment:
Is this the only change, compared to the overriden method? Can this be moved to `write` to avoid overriding both?

Reply:
Good point. Maybe we complete reuse the collect(...) super method and just do a bytes ref conversation in write(...) method so that `FlattenedFieldSyntheticWriterHelper` can use the collected values.